internal/sbi/producer/nf_management.go

```diff
@@ -230,10 +230,12 @@ func RemoveSubscriptionProcedure(subscriptionID string) {
 }
 
 func GetNFInstancesProcedure(nfType string, limit int) (*nrf_context.UriList, *models.ProblemDetails) {
-	// nfType := c.Query("nf-type")
-	// limit, err := strconv.Atoi(c.Query("limit"))
 	collName := "urilist"
 	filter := bson.M{"nfType": nfType}
+	if nfType == "" {
+		// if query parameter is not append, the nftype defalue doesn't use.
+		filter = bson.M{}
```

Comment:
Have you tested this? I'm not sure this will work, because the NFs in `urilist` are grouped by `nfType`. That means if you use `ongoapi.RestfulAPIGetOne(collName, filter)`, you only get the NF instances for _one_ (arbitrary) NF type.

I think to solve the problem, you either need to get the data from a different collection (maybe `NfProfile`, but I'm not sure about this, I do not know this project well enough) or you need to get multiple documents from `urilist` and concatenate the lists. In any case, I think you need a different function than `RestfulAPIGetOne`, because that will only give you _one_ document.

That said, I'm not particularly experienced with MongoDB and the data structures that free5GC uses here, so I may be wrong about this.